riscv/mmu.h

```diff
@@ -141,10 +141,12 @@ class mmu_t
   // template for functions that perform an atomic memory operation
   template<typename T, typename op>
   T amo(reg_t addr, op f) {
+    check_triggers(triggers::OPERATION_AMO, addr);
     convert_load_traps_to_store_traps({
       store_slow_path(addr, sizeof(T), nullptr, 0, false, true);
       auto lhs = load<T>(addr);
       store<T>(addr, f(lhs));
```

Comment:
Similarly, isn't it possible that `store` will accidentally hit the data trigger, therefore allowing the load side effect to occur without actually performing the store?